dts/bindings/mtd/partition.yaml

```diff
@@ -16,6 +16,17 @@ properties:
 child-binding:
     description: Flash partition child node
     properties:
+       mntauto:
+          type: boolean
+          required: false
+          description: if the partition should be mounted or not
+       mnttype:
```

Comment:
This is fragile (but explains why the type changed).  If a new file system is integrated into Zephyr then the offset for external ones will change, and the devicetree will select the wrong type.

I think this is a showstopper for this feature, unless we extend support for filesystem types to support a string identifier with a clear naming policy (e.g. "FatFs", "littlefs", "ext:myfs", ...) that can be used here instead.

Reply:
I think it would be possible, I have an idea for that that seems to work on simpler prototype and it will probably allow me to retreat from changing the `FS_*` definitions of file systems.